sdks/python/src/opik/decorator/base_track_decorator.py

```diff
@@ -180,6 +187,9 @@ def wrapper(*args, **kwargs) -> Any:  # type: ignore
                     output=result,
                     capture_output=capture_output,
                 )
+                if flush:
```

Comment:
@dsblank this will not work for generators. See the code above, if the generator was decorated, we return wrapped generator before you are calling `flush()`.

Reply:
Is there anything we can do to flush after the generator is done?

Reply:
we can do flushing in `_after_call` method.